### PR TITLE
edited meat item modal

### PR DESCRIPTION
**Pull request by Toriong**

### CODE REVIEW

File: src/components/AddOnItem.js [hunk 1]
```diff
@@ -1,57 +1,53 @@
...
-        let addOns;
-        if (order.addOns) {
-            addOns = [...order.addOns, addOnItem.id]
-        } else {
-            addOns = [addOnItem.id]
-        }
-        setOrder({
-            ...order,
-            addOns: addOns
+        const addOns_ = meatItem.addOns ? [...meatItem.addOns, addOnItem.id] : [addOnItem.id]
```
Comment (ilyador) [inline src/components/AddOnItem.js]: Way better!
One thing though, the `_` character that represents temp variables should be in the beginning and not the end.

File: src/components/AddOnItem.js [hunk 2]
```diff
@@ -1,57 +1,53 @@
...
 
     const deleteAddOnFromOrder = () => {
-        const addOns = order.addOns.filter((addOnId) => addOnId !== addOnItem.id)
-        setOrder({
-            ...order,
-            addOns: addOns
-        });
+        const addOns_ = meatItem.addOns.filter(addOnId => addOnId !== addOnItem.id);
+        if (!addOns_.length) {
+            delete meatItem.addOns
```
Comment (ilyador) [inline src/components/AddOnItem.js]: Why are you deleting the addons?
If they are empty you can check for their length elsewhere.

File: src/components/AddOnItem.js [hunk 3]
```diff
@@ -1,57 +1,53 @@
...
     useEffect(() => {
-        if (order.addOns) {
-            order.addOns.forEach((addOnId) => {
-                if (addOnId === addOnItem.id) {
-                    setBoxClicked(true);
-                };
-            });
-        };
-    }, []);
+        computeOrderPrice()
```
Comment (ilyador) [inline src/components/AddOnItem.js]: It seems like a bad pattern to use a prop function as an effect.
Have you made sure it is not relying on a state that can change and cause closure bugs?

File: src/components/AddOnItem.js [hunk 4]
```diff
@@ -1,57 +1,53 @@
...
 import '../css/addOnItem.css'
 
 
-
-
-
-const AddOnItem = ({ addOnItem, order, setOrder, computeOrderTotalPrice }) => {
-
+const AddOnItem = ({ addOnItem, meatItem, computeOrderPrice, setMeatItem }) => {
     const [boxClicked, setBoxClicked] = useState(false);
```
Comment (ilyador) [inline src/components/AddOnItem.js]: I can't figure out what does `boxClicked` mean.
The name is too generic.

File: src/components/AddOnItem.js (hunk 3 above)
Comment (Toriong) [reply src/components/AddOnItem.js]: the function `computeOrderPrice` does have the state value of `orderCount` within it. But for the useEffect that it is located in, it has `isAddOnOnOrder` (the new name for `boxClicked`) set as its dependency. I can find another way to compute the price of the order when an add-on is added?

File: src/components/MeatItemModal.js [hunk 5]
```diff
@@ -48,6 +49,12 @@ const MeatItemModal = ({ order, setOrder, isCartButtonsOnModal, setIsMeatItemMod
     const updateOrder = () => {
         const updatedCartOrders = cartOrders.orders.map(cartOrder_ => {
             if (cartOrder_.orderId === order.orderId) {
+                if (order.addOns) {
+                    if (!order.addOns.length) {
```
Comment (ilyador) [inline src/components/MeatItemModal.js:53]: I think we talked about this section and the purpose of the delete, that addons shouldn't be there in the first place.
Also, this is not how you write a double condition, you use `&&`

File: src/css/cart.css [hunk 6]
```diff
@@ -114,6 +112,7 @@
   width: 250px;
   height: 50px;
   background: black;
+  border-radius: 2.5vh;
```
Comment (ilyador) [inline src/css/cart.css:115]: Did you connect on purpose the border radius to the display size?

**Review by ilyador — COMMENTED** [review]
We talked about a lot of things last time and I expected to see more changes, especially since it took you so long.

**Pull request merged**